### PR TITLE
Adding content to the SOSS section

### PR DESCRIPTION
**Pull request by mxgrey**

### CODE REVIEW

File: src/soss.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # SOSS
 
-The system-of-systems synthesizer (SOSS) is a tool which provides protocol
-translation between different subsystems. Currently, ROS 2, ROS 1, and
-Websockets are supported. Such composite systems can be called The ROS-SOSS.
+The ecosystem of different message passing systems is vast and diverse. Without any one single system that is unanimously considered the best for all applications, we are left to consider how we can tie together disparate message passing systems to bridge the gap between the different kinds of applications that all play a critical role in a modern, intelligent robotics solution. The best protocols to use for inter-robot communication might not be the best for remote operator communication or for end-user communication.
```
Comment (mdfesto) [inline src/soss.md:7]:
```suggestion
The system-of-systems synthesizer (SOSS) is a tool which provides protocol translation between different subsystems. Such composite systems can be called The ROS-SOSS. To see the current implementation status, see the [SOSS repository](https://github.com/osrf/soss). 

##Motivation and Introduction

The ecosystem of different message passing systems is vast and diverse. Without any one single system that is unanimously considered the best for all applications, we are left to consider how we can tie together disparate message passing systems to bridge the gap between the different kinds of applications that all play a critical role in a modern, intelligent robotics solution. The best protocols to use for inter-robot communication might not be the best for remote operator communication or for end-user communication.
```

Comment (mdfesto) [reply src/soss.md:7]: The other sections typically have an overview header and then some details so I've re-worded the overview and inserted a section title here. Also we don't have any link to the SOSS repo so this seemed like a good place to put it.

File: src/soss.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
+
+The base SOSS package also provides some cmake tools to help with auto-generating message definitions for message-passing systems that require static message definitions at compile-time. For message-passing systems with dynamic message types, the plugin can automatically take care of the translation at runtime, so auto-generation is unnecessary.
+
+For a deeper look at how to use SOSS, we recommend reading [the documentation provided by eProsima](https://integration-service.docs.eprosima.com/en/latest/).
```
Comment (mdfesto) [inline src/soss.md]: I'm not sure if you wanted to point to it but the link is to the "eProsima Integration-Service" which "_is a tool based on SOSS that allows intercommunicating any DDS-based system with any other protocol_". 

I think you meant to link to their SOSS Documentation (https://soss.docs.eprosima.com/en/latest/index.html)?

**Review by mdfesto — COMMENTED** [review]
Thanks for summarizing this @mxgrey ! Made a few comments for your review

File: src/soss.md (hunk 2 above)
Comment (mxgrey) [reply src/soss.md]: Oh, good call. It looked the same to me, so I didn't realize there was a difference.

**Pull request merged**